user-interface/src/main/java/life/qbic/datamanager/views/projects/project/measurements/registration/MeasurementUpload.java

```diff
@@ -0,0 +1,424 @@
+    uploadItemsDisplay.addFileToDisplay(measurementFileDisplay);
+    refreshFileItemDisplay();
+  }
+
+
+  private void onFileRejected(FileRejectedEvent fileRejectedEvent) {
+    String errorMessage = fileRejectedEvent.getErrorMessage();
+    displayError(errorMessage);
+  }
+
+  private ParsingResult parseXLSX(InputStream inputStream) {
+    return XLSXParser.create().parse(inputStream);
+  }
+
+  private ParsingResult parseTSV(InputStream inputStream) {
+    return TSVParser.create().parse(inputStream);
+  }
+
+  @Override
+  public InputValidation validate() {
+    var inputWithFailureSearch = measurementFileItems.stream().map(MeasurementFileItem::measurementValidationReport)
+        .map(MeasurementValidationReport::validationResult)
+        .filter(ValidationResult::containsFailures).findAny();
+    if (inputWithFailureSearch.isEmpty()) {
+      return InputValidation.passed();
+    }
+    return InputValidation.failed();
+  }
+
+  @Override
+  public boolean hasChanges() {
+    return !measurementFileItems.isEmpty();
+  }
+
+  private static class UploadItemsDisplay extends Div {
+
+    private final Div uploadSection;
+    private final Div uploadedItemsSection;
+    private final Div uploadedItemsDisplays;
+
+    public UploadItemsDisplay(Upload upload) {
+      var uploadSectionTitle = new Span("Upload the measurement data");
+      uploadSectionTitle.addClassName("section-title");
+
+      var saveYourFileInfo = new InfoBox().setInfoText(
+              "When uploading a tab-separated file, please save your Excel file as UTF-16 Unicode Text (*.txt) before uploading.")
+          .setClosable(false);
+
+      var restrictions = new Div();
+      restrictions.addClassName("restrictions");
+      restrictions.add(new Span("Supported file formats: .txt, .tsv, .xlsx"));
+      restrictions.add(
+          "Maximum file size: %s MB".formatted(MAX_FILE_SIZE_BYTES / Math.pow(1024, 2)));
+
+      this.uploadSection = new Div();
+      uploadSection.add(uploadSectionTitle, saveYourFileInfo, upload, restrictions);
+      uploadSection.addClassName("upload-section");
+
+      uploadedItemsSection = new Div();
+      uploadedItemsSection.addClassName("uploaded-items-section");
+
+      var uploadedItemsSectionTitle = new Span("Uploaded files");
+      uploadedItemsSectionTitle.addClassName("section-title");
+
+      uploadedItemsDisplays = new Div();
+      uploadedItemsDisplays.addClassName("uploaded-measurement-items");
+      uploadedItemsSection.add(uploadedItemsSectionTitle, uploadedItemsDisplays);
+      add(uploadSection, uploadedItemsSection);
+      addClassName("upload-items-display");
+    }
+
+    private void addFileToDisplay(MeasurementFileDisplay measurementFileDisplay) {
+      uploadedItemsDisplays.add(measurementFileDisplay);
+    }
+
+    private void removeFileFromDisplay(String fileName) {
+      MeasurementFileDisplay[] fileDisplays = fileDisplaysWithFileName(fileName);
```

Comment:
can there be multiple with the same filename?

Reply:
technically yes